### DISCUSSION TITLE
Bulk delete

### DISCUSSION THREAD

**Discussion by stychu** [supabase/supabase#2436] [Questions] [answered]
From given examples it's not clear to me if it's possible to do bulk delete of records or do I have to do I one by one?

```
const { data, error } = await postgrest
  .from('cities')
  .delete()
  .match({ id: 666 })
  ```
  

Is this possible?
  ```
  const { data, error } = await postgrest
  .from('cities')
  .delete()
  .match([
  { id: 666 },
    { id: 444}
      { id: 333 }
        { id: 222 }
  ])
  ```

**Comment by dragarcia** [comment, marked as answer]
Hi @stychu 👋 

You can use the [`.in()` filter](https://supabase.io/docs/reference/javascript/in#with-delete) to list down the ids in a single array and send it for bulk deletion. Using your example, it would look something like:
```
const { data, error } = await postgrest
.from('cities')
.delete()
.in('id', [666, 444, 333, 222])
```
